src/main/java/net/machpi/runelite/influxdb/InfluxDbPlugin.java

```diff
@@ -73,16 +74,15 @@ InfluxDbConfig getConfig(ConfigManager configManager) {
      * Don't use a shared executor because we don't want to block any game threads.
      */
     private final ScheduledExecutorService executor = new ExecutorServiceExceptionLogger(Executors.newSingleThreadScheduledExecutor());
-
-    private boolean loginFlag;
-    private int loggedInTick = Integer.MAX_VALUE;
+    private final EnumMap<Skill, Integer> previousStatXp = new EnumMap<>(Skill.class);
+    private GameState prevGameState;
 
     @Subscribe
     public void onStatChanged(StatChanged statChanged) {
-        // stat change is kicked off when logged in. only update stats when init'ing stats has finished
-        if (loggedInTick >= client.getTickCount()) {
+        int prevStatXp = previousStatXp.getOrDefault(statChanged.getSkill(), 0);
+        if (statChanged.getXp() == prevStatXp)
             return;
-        }
+        previousStatXp.put(statChanged.getSkill(), statChanged.getXp());
```

Comment:
This is still triggering xp updates upon login. And sending fake slayer activity.